docs/userguide/migration.rst

```diff
@@ -287,6 +287,117 @@ After::
 
         await to_thread.run_sync(wrapper)
 
+Signals and events
+------------------
+
+In support of `structured concurrency`_, the signalling system (not to be confused with
+operating system signals like ``SIGTERM`` et al), has been refactored to require the use
+of context managers wherever possible.
+
+Migrating custom event classes
+++++++++++++++++++++++++++++++
+
+As the :class:`Event` class no longer has an initializer, you need to remove the
+``source`` and ``topic`` initializer parameters from your own subclasses, and drop the
+``super().__init__(source, topic)`` call. You may also want to take this opportunity to
+refactor them into data classes.
+
+Before::
+
+    from asphalt.core import Signal, Event
+
+    class MyEvent(Event):
+        def __init__(self, source, topic, an_attribute: str):
+            super().__init__(source, topic)
+            self.an_attribute = an_attribute
+
+After::
+
+    from dataclasses import dataclass
+
+    from asphalt.core import Signal, Event
+
+    @dataclass
+    class MyEvent(Event):
+        an_attribute: str
+
+Iterating over events
++++++++++++++++++++++
+
+As the ``connect()`` and ``disconnect()`` signal methods have been eliminated, you need
+to use the :meth:`Signal.stream_events` method or
```

Comment:
```suggestion
to use the :meth:`Signal.stream_events` method.
```

Reply:
That unfinished sentence was actually meant to also mention the free `stream_events()` function

Reply:
Updated.